src/equations/shallow_water_multilayer_1d.jl

```diff
@@ -0,0 +1,543 @@
+                                 equations::ShallowWaterMultiLayerEquations1D)
+
+Create a boundary state by reflecting the normal velocity component and keep
+the tangential velocity component unchanged. The boundary water height is taken from
+the internal value.
+
+For details see Section 9.2.5 of the book:
+- Eleuterio F. Toro (2001)
+  Shock-Capturing Methods for Free-Surface Shallow Flows
+  1st edition
+  ISBN 0471987662
+"""
+@inline function Trixi.boundary_condition_slip_wall(u_inner, orientation_or_normal,
+                                                    direction,
+                                                    x, t, surface_flux_function,
+                                                    equations::ShallowWaterMultiLayerEquations1D)
+    # Create the "external" boundary solution state
+    h = waterheight(u_inner, equations)
+    hv = momentum(u_inner, equations)
+    b = u_inner[end]
+
+    u_boundary = SVector(h..., -hv..., b)
+
+    # Calculate the boundary flux
+    if iseven(direction) # u_inner is "left" of boundary, u_boundary is "right" of boundary
+        f = surface_flux_function(u_inner, u_boundary, orientation_or_normal, equations)
+    else # u_boundary is "left" of boundary, u_inner is "right" of boundary
+        f = surface_flux_function(u_boundary, u_inner, orientation_or_normal, equations)
+    end
+    return f
+end
+
+# Calculate 1D flux for a single point
+@inline function Trixi.flux(u, orientation::Integer,
+                            equations::ShallowWaterMultiLayerEquations1D)
+    # Extract waterheight and momentum and compute velocities
+    hv = momentum(u, equations)
+    v = velocity(u, equations)
+
+    # Initialize flux vector
+    f = zero(MVector{2 * nlayers(equations) + 1, real(equations)})
+    # Calculate fluxes in each layer
+    for i in eachlayer(equations)
+        f_h = hv[i]
+        # The momentum flux simplifies as the pressure is included in the nonconservative term
+        f_hv = hv[i] * v[i]
+
+        setlayer!(f, f_h, f_hv, i, equations)
+    end
+    return SVector(f)
+end
+
+"""
+    flux_nonconservative_ersing_etal(u_ll, u_rr, orientation::Integer,
+                                     equations::ShallowWaterTwoLayerEquations1D)
+
+Non-symmetric path-conservative two-point flux discretizing the nonconservative (source) term
+that contains the gradients of the bottom topography and waterheights from the coupling between layers
+and the nonconservative pressure formulation [`ShallowWaterMultiLayerEquations1D`](@ref).
+
+When the bottom topography is nonzero this scheme will be well-balanced when used with [`flux_ersing_etal`](@ref).
+
+In the two-layer setting this combination is equivalent to the fluxes in:
+- Patrick Ersing, Andrew R. Winters (2023)
+  An entropy stable discontinuous Galerkin method for the two-layer shallow water equations on 
+  curvilinear meshes
+  [DOI: 10.48550/arXiv.2306.12699](https://doi.org/10.48550/arXiv.2306.12699)
```

Comment:
```suggestion
  [DOI: 10.1007/s10915-024-02451-2](https://doi.org/10.1007/s10915-024-02451-2)
```
now that this is published we can point to the official DOI for the JSC version.